ObjectiveGit/GTRepository.h

```diff
@@ -209,6 +231,18 @@ typedef NS_ENUM(NSInteger, GTRepositoryStateType) {
 /// Returns the initialized repository, or nil if an error occurred.
 - (nullable instancetype)initWithURL:(NSURL *)localFileURL error:(NSError **)error;
 
+/// Convenience initializer to find and open a repository with extended controls.
+///
+/// localFileURL - The file URL for the new repository. Cannot be nil.
+/// flags        - A combination of the `GTRepositoryOpenFlags` flags.
+/// ceilingDirs  - A GIT_PATH_LIST_SEPARATOR delimited list of path prefixes at
+///                which the search for a containing repository should terminate.
+///                Can be NULL.
+/// error        - The error if one occurs.
+///
+/// Returns the initialized repository, or nil if an error occurred.
+- (nullable instancetype)initWithURL:(NSURL *)localFileURL flags:(UInt32)flags ceilingDirs:(nullable const char *)ceilingDirs error:(NSError **)error;
```

Comment:
Actually, is there any reason you would rather drop the instance initializer rather than the class initializer? Keeping the convenience instance initializer allows my Swift code to look cleaner: `GTRepository(url: url, flags: 0, ceilingDirs: nil)`